stdlib/2and3/builtins.pyi

```diff
@@ -1189,7 +1189,9 @@ else:
 @overload
 def iter(__iterable: Iterable[_T]) -> Iterator[_T]: ...
 @overload
-def iter(__function: Callable[[], _T], __sentinel: _T) -> Iterator[_T]: ...
+def iter(__function: Callable[[], Optional[_T]], __sentinel: None) -> Iterator[_T]: ...
+@overload
+def iter(__function: Callable[[], Any], __sentinel: Any) -> Iterator[Any]: ...
```

Comment:
the `Any` is too relaxed here. the output from the callable will always be the same as the value yielded by the iterator, regardless of what the sentinel type is. i will open a pull request to make this more strict.

Reply:
#3326